### PR TITLE
Cleanup NVTX Markers for dynamic shape profiling

### PR DESCRIPTION
**Pull request by csarofeen**
Cleanup nvtx markers for dynamic latencies: remove those that weren't significant, add others that were, make schedulers consistent.

Stacked on https://github.com/NVIDIA/Fuser/pull/2884

### CODE REVIEW

File: csrc/kernel_cache.cpp [hunk 1]
```diff
@@ -692,8 +692,10 @@ DynamicTransformInitialInfo& FusionExecutorCache::initialInfo() {
 // since they both involve re-segmentation and re-compilation of the Fusion.
 FusionKernelRuntime* FusionExecutorCache::getKernelRuntimeFor(
     const KernelArgumentHolder& args,
-    std::optional<PrimDataType> forced_index_type) {
```
Comment (csarofeen) [inline csrc/kernel_cache.cpp:695]: Missing {

**Review by liqiangxl — APPROVED** [review]
LGTM.

File: csrc/executor.cpp [hunk 2]
```diff
@@ -1039,7 +1037,7 @@ std::vector<at::Tensor> allocateOutputs(
     const std::vector<FusionExecutor::GlobalBufferInfo>& output_info,
     const c10::Device& device,
     ExpressionEvaluator& ee) {
-  FUSER_PERF_SCOPE("allocateOutputs");
+  FUSER_PERF_SCOPE("executor.cpp::allocateOutputs");
```
[the remark this answers is not in the document]
Comment (csarofeen) [reply csrc/executor.cpp:1040]: It's not part of the class, so I named it after the file.

File: csrc/kernel_cache.cpp [hunk 3]
```diff
@@ -773,6 +774,7 @@ FusionKernelRuntime* FusionExecutorCache::getKernelRuntimeFor(
   }
 
   if (!reusing) {
+    FUSER_PERF_SCOPE("FusionExecutorCache::getKernelRuntimeFor::!reusing");
```
[the remark this answers is not in the document]
Comment (csarofeen) [reply csrc/kernel_cache.cpp:777]: Will leave to the future, don't want to delay this PR as we're trying to wrap up dynamic latencies for a use case.

**Pull request merged**